packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts

```diff
@@ -29,48 +31,110 @@ import {
+import { IErrorBase } from "@fluidframework/core-interfaces";
+// eslint-disable-next-line import/no-internal-modules
+import { RetriableSummaryError } from "@fluidframework/container-runtime/dist/summary/summaryGenerator.js";
+// eslint-disable-next-line import/no-internal-modules
+import { defaultMaxAttemptsForSubmitFailures } from "@fluidframework/container-runtime/dist/summary/runningSummarizer.js";
+// eslint-disable-next-line import/no-internal-modules
+import { ISweepMessage } from "@fluidframework/container-runtime/dist/gc/gcDefinitions.js";
+import {
+	getGCDeletedStateFromSummary,
+	getGCStateFromSummary,
+	manufactureHandle,
+} from "./gcTestSummaryUtils.js";
+
+/**
+ * Validates that the given data store state is correct in the summary.
+ * If expectDelete is true:
+ * - It should be deleted from the data store summary tree.
+ * If expectGCStateHandle is not true:
+ * - It should not be present in the GC state in GC summary tree.
+ * - It should be present in the deleted nodes in GC summary tree.
+ */
+function validateDataStoreStateInSummary(
+	summaryTree: ISummaryTree,
+	dataStoreNodePath: string,
+	expectDelete: boolean = true,
+	expectGCStateHandle: boolean = false,
+) {
+	const shouldShouldNot = expectDelete ? "should" : "should not";
+
+	// Check if the data store is deleted from the data store summary tree or not.
+	const deletedDataStoreId = dataStoreNodePath.split("/")[1];
+	const channelsTree = (summaryTree.tree[channelsTreeName] as ISummaryTree).tree;
+	assert.notEqual(
+		Object.keys(channelsTree).includes(deletedDataStoreId),
+		expectDelete,
+		`Data store ${deletedDataStoreId} ${shouldShouldNot} have been deleted from the summary`,
+	);
+
+	if (expectGCStateHandle) {
+		assert.equal(
+			summaryTree.tree[gcTreeKey].type,
+			SummaryType.Handle,
+			"Expecting the GC tree to be handle",
+		);
+		return;
+	}
+
+	// Validate that the GC state does not contain an entry for the deleted data store.
+	const gcState = getGCStateFromSummary(summaryTree);
+	assert(gcState !== undefined, "GC tree is not available in the summary");
+	assert.notEqual(
+		Object.keys(gcState.gcNodes).includes(dataStoreNodePath),
+		expectDelete,
+		`Data store ${dataStoreNodePath} ${shouldShouldNot} have been removed from GC state`,
+	);
+
+	// Validate that the deleted nodes in the GC data has the deleted data store.
+	const deletedNodesState = getGCDeletedStateFromSummary(summaryTree);
+	assert.equal(
+		deletedNodesState?.includes(dataStoreNodePath) ?? false,
+		expectDelete,
+		`Data store ${dataStoreNodePath} ${shouldShouldNot} be in deleted nodes`,
+	);
+}
 
 /**
  * These tests validate that SweepReady data stores are correctly swept. Swept datastores should be
  * removed from the summary, added to the GC deleted blob, and prevented from changing (sending / receiving ops,
  * loading, etc.).
  */
 describeCompat("GC data store sweep tests", "NoCompat", (getTestObjectProvider) => {
-	const remainingTimeUntilSweepMs = 100;
```

Comment:
It looks like we got rid of the tests that were single container loading right before any timeouts. Do we still have coverage somewhere?

Reply:
I didn't see those tests in this file. I assumed that this part I removed was carry over from other tests where it is actually used.

Reply:
Ok, as long as we have coverage, I'm good